src/classes/RecipeRepository.js

```diff
@@ -23,35 +17,41 @@ class RecipeRepository {
       return acc;
     }, []);
     recipesWithTags.forEach(recipe => {
-      this.recipesToShow.push(recipe)
+      this.recipesToShow.push(recipe);
     })
   }
 
-  getIngredientID(ingName, ingredientsData) {
+  getIngredientIDs(ingName, ingredientsData) {
     const ingNameLC = ingName.toLowerCase();
-    const ingredientIDFromSearch = ingredientsData.reduce((acc, ingredient) => {
-      if (ingredient.name === ingNameLC) {
-        acc = ingredient.id;
-      };
+    const ingredientIDsFromSearch = ingredientsData.reduce((acc, ingredient) => {
+      const ingredientName = ingredient.name.toLowerCase();
+      if (ingredientName.includes(ingNameLC)) {
+        acc.push(ingredient.id);
+      }
       return acc;
-    }, 0);
-    return ingredientIDFromSearch;
+    }, []);
+    return ingredientIDsFromSearch;
   }
 
   filterByIng(nameOrIng, ingredientsData, recipes) {
-    const ingredientID = this.getIngredientID(nameOrIng, ingredientsData);
-    return recipes.filter((recipe) => {
-      let ingIDs = recipe.ingredients.map((ingredient) => {
-        return ingredient.id;
-      });
-      return ingIDs.includes(ingredientID);
-    });
+    const ingredientIDs = this.getIngredientIDs(nameOrIng, ingredientsData);
```

Comment:
YES! This allows us to put in a word like flour and it will return anything with flour as an ingredient. Before, you'd have to specifically search 'wheat flour' etc for anything to pop. Now, a recipe with any word of the ingredient will populate.